pkg/common/config/config.go

```diff
@@ -442,6 +442,10 @@ func GetGCconfig(ctx context.Context, cfgPath string) (*Config, error) {
 			return cfg, err
 		}
 	}
+	// Feature states config info is not provided in vsphere conf in tkg clusters. Initializing with default values
```

Comment:
PVCSI does not use vsphere conf. It uses cns-csi.conf(see https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/manifests/dev/vsphere-7.0u1/guestcluster/1.17/pvcsi.yaml#L355). Can the FSS configuration be added to the cns_csi.conf secret? And are these SV cluster FSS config map? 

Btw, can you explain how was all this working without this PR?

Reply:
> Can the FSS configuration be added to the cns_csi.conf secret? And are these SV cluster FSS config map?

FSS configuration cannot be added to cns_csi.conf secret. Based on the design discussion with TKG folks, it will impose TKG controllers to know where pvCSI reads the config and editing an existing config secret is something that was not agreed by them.  Hence FSS configuration is kept as a separate config-map.
Yes, these are SV cluster FSS config map information which pvCSI features needs to know.

> Btw, can you explain how was all this working without this PR?

Prior to this PR, there is a regression in TKG while loading FSS. This regression was introduced while making the FSS configurable for Vanilla. By mistake, I had kept the below logic(Line: 446, 447) in `validateConfig()`. But `validateConfig()` is only invoked for CNS-CSI and Vanilla flavors.
Prior to introducing the regression, the FSS was not present in Vanilla and hence everything(config-map name, namespace) was initialized to `csi-feature-states` and `vmware-system-csi`. Hence things were working fine before.

Reply:
>FSS configuration cannot be added to cns_csi.conf secret

I mean, can the  SV FSS configmap name and namespace be added as a property in cns-csi.conf? I see that your code is hard-coding it and it will make it difficult to use PVCSI outside of TKG use cases.

Reply:
`cns-csi.conf` contains information used by guest cluster to access supervisor cluster endpoint. IMO, keeping FSS information here would mix things up. 
Overall, I agree with you that we need to make this configurable for use cases outside of TKGS.
And also, introducing a template variable now will need co-ordination with TKG folks.

I would like to work on this as a separate work-item: https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/404

Reply:
Wouldn't it be just a matter of adding these properties in the pvcsi-config ConfigMap? 

```
  sv-fss-configmap-name = "csi-feature-states"
  sv-fss-configmap-namespace = "{{ .PVCSINamespace }}"
```

This should just work, right?

Reply:
Makes sense. Addressed this change in the latest commit

Reply:
This is how the pvcsi-config configmap looks:

<pre>
apiVersion: v1
data:
  cns-csi.conf: |
    [GC]
    endpoint = "supervisor.default.svc"
    port = "6443"
    tanzukubernetescluster-uid = "3bedb18d-b662-4600-ad95-296a544a6c9d"
    tanzukubernetescluster-name = "test-cluster-e2e-script"
    [FeatureStatesConfig]
    name = "csi-feature-states"
    namespace = "vmware-system-csi"
kind: ConfigMap
metadata:
  creationTimestamp: "2020-10-01T03:59:26Z"
  name: pvcsi-config
  namespace: vmware-system-csi
  resourceVersion: "240337"
  selfLink: /api/v1/namespaces/vmware-system-csi/configmaps/pvcsi-config
  uid: 6a9b1337-bbb6-4742-8369-713260d4ddf9
</pre>



FSS loaded successfully for syncer and driver containers:

<pre>
kubectl logs vsphere-csi-controller-74665d87b7-64l9h -n vmware-system-csi -c vsphere-syncer
{"level":"info","time":"2020-10-01T18:27:41.588443073Z","caller":"logger/logger.go:37","msg":"Setting default log level to :\"PRODUCTION\""}
{"level":"info","time":"2020-10-01T18:27:41.591330859Z","caller":"syncer/main.go:62","msg":"Version : 147b18bf-dirty","TraceId":"2eeaa771-c341-4fe4-b6cd-6db383328768"}
{"level":"info","time":"2020-10-01T18:27:41.59175287Z","caller":"syncer/main.go:75","msg":"Starting container with operation mode: METADATA_SYNC","TraceId":"2eeaa771-c341-4fe4-b6cd-6db383328768"}
{"level":"info","time":"2020-10-01T18:27:41.596222844Z","caller":"kubernetes/kubernetes.go:83","msg":"k8s client using in-cluster config","TraceId":"2eeaa771-c341-4fe4-b6cd-6db383328768"}
{"level":"info","time":"2020-10-01T18:27:41.597876563Z","caller":"kubernetes/kubernetes.go:267","msg":"Setting client QPS to 100.000000 and Burst to 100.","TraceId":"2eeaa771-c341-4fe4-b6cd-6db383328768"}
I1001 18:27:41.615012       1 leaderelection.go:242] attempting to acquire leader lease  vmware-system-csi/vsphere-syncer...
I1001 18:28:34.373679       1 leaderelection.go:252] successfully acquired lease vmware-system-csi/vsphere-syncer
{"level":"info","time":"2020-10-01T18:28:34.387175939Z","caller":"syncer/metadatasyncer.go:116","msg":"Initializing MetadataSyncer"}
{"level":"info","time":"2020-10-01T18:28:34.397276024Z","caller":"kubernetes/kubernetes.go:83","msg":"k8s client using in-cluster config"}
{"level":"info","time":"2020-10-01T18:28:34.403566605Z","caller":"kubernetes/kubernetes.go:267","msg":"Setting client QPS to 100.000000 and Burst to 100."}
{"level":"info","time":"2020-10-01T18:28:34.414811707Z","caller":"k8sorchestrator/k8sorchestrator.go:49","msg":"Initializing k8sOrchestratorInstance"}
{"level":"info","time":"2020-10-01T18:28:34.415322166Z","caller":"kubernetes/kubernetes.go:83","msg":"k8s client using in-cluster config"}
{"level":"info","time":"2020-10-01T18:28:34.418290615Z","caller":"kubernetes/kubernetes.go:267","msg":"Setting client QPS to 100.000000 and Burst to 100."}
{"level":"info","time":"2020-10-01T18:28:34.439068833Z","caller":"k8sorchestrator/k8sorchestrator.go:139","msg":"New feature states values stored successfully: map[volume-extend:true volume-health:true]"}
{"level":"info","time":"2020-10-01T18:28:34.441917635Z","caller":"k8sorchestrator/k8sorchestrator.go:77","msg":"k8sOrchestratorInstance initialized"}
{"level":"info","time":"2020-10-01T18:28:34.442833999Z","caller":"kubernetes/kubernetes.go:267","msg":"Setting client QPS to 100.000000 and Burst to 100."}
{"level":"info","time":"2020-10-01T18:28:34.466007326Z","caller":"k8sorchestrator/k8sorchestrator.go:139","msg":"New feature states values stored successfully: map[volume-extend:true volume-health:true]","TraceId":"9998b59b-d03b-4f07-b366-1216ff2965b1"}





kubectl logs vsphere-csi-controller-74665d87b7-64l9h -n vmware-system-csi -c vsphere-csi-controller
{"level":"info","time":"2020-10-01T18:27:41.076416797Z","caller":"vsphere-csi/main.go:42","msg":"Version : 147b18bf-dirty"}
{"level":"info","time":"2020-10-01T18:27:41.084101805Z","caller":"logger/logger.go:37","msg":"Setting default log level to :\"PRODUCTION\""}
{"level":"info","time":"2020-10-01T18:27:41.096067842Z","caller":"wcpguest/controller.go:80","msg":"Initializing WCPGC CSI controller","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}
{"level":"info","time":"2020-10-01T18:27:41.100099391Z","caller":"kubernetes/kubernetes.go:267","msg":"Setting client QPS to 50.000000 and Burst to 50.","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}
{"level":"info","time":"2020-10-01T18:27:41.100208877Z","caller":"kubernetes/kubernetes.go:135","msg":"Connecting to supervisor cluster using the certs/token in Guest Cluster config","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}
{"level":"info","time":"2020-10-01T18:27:41.445916904Z","caller":"k8sorchestrator/k8sorchestrator.go:49","msg":"Initializing k8sOrchestratorInstance","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}
{"level":"info","time":"2020-10-01T18:27:41.44648518Z","caller":"kubernetes/kubernetes.go:83","msg":"k8s client using in-cluster config","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}
{"level":"info","time":"2020-10-01T18:27:41.448898664Z","caller":"kubernetes/kubernetes.go:267","msg":"Setting client QPS to 50.000000 and Burst to 50.","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}
{"level":"info","time":"2020-10-01T18:27:41.475616401Z","caller":"k8sorchestrator/k8sorchestrator.go:139","msg":"New feature states values stored successfully: map[volume-extend:true volume-health:true]","TraceId":"dd493b28-9ed9-48f4-86a2-e20737248260"}





</pre>